data/pc/1.6.4/protocol.json

```diff
@@ -0,0 +1,2286 @@
+            "type": "f32"
+          }
+        ]
+      ],
+      "packet_world_event": [
+        "container",
+        [
+          {
+            "name": "effectId",
+            "type": "i32"
+          },
+          {
+            "name": "position",
+            "type": "position_ibi"
+          },
+          {
+            "name": "data",
+            "type": "i32"
+          },
+          {
+            "name": "global",
+            "type": "bool"
+          }
+        ]
+      ],
+      "packet_named_sound_effect": [
+        "container",
+        [
+          {
+            "name": "soundName",
+            "type": "string"
+          },
+          {
+            "name": "position",
+            "type": "position_iii"
+          },
+          {
+            "name": "volume",
+            "type": "f32"
+          },
+          {
+            "name": "pitch",
+            "type": "i8"
+          }
+        ]
+      ],
+      "packet_world_particles": [
+        "container",
+        [
+          {
+            "name": "particleName",
+            "type": "string"
+          },
+          {
+            "name": "x",
+            "type": "f32"
+          },
+          {
+            "name": "y",
+            "type": "f32"
+          },
+          {
+            "name": "z",
+            "type": "f32"
+          },
+          {
+            "name": "offsetX",
+            "type": "f32"
+          },
+          {
+            "name": "offsetY",
+            "type": "f32"
+          },
+          {
+            "name": "offsetZ",
+            "type": "f32"
+          },
+          {
+            "name": "particleSpeed",
+            "type": "f32"
+          },
+          {
+            "name": "particles",
+            "type": "i32"
+          }
+        ]
+      ],
+      "packet_game_state_change": [
+        "container",
+        [
+          {
+            "name": "reason",
+            "type": "i8"
+          },
+          {
+            "name": "gameMode",
+            "type": "i8"
+          }
+        ]
+      ],
+      "packet_spawn_entity_weather": [
+        "container",
+        [
+          {
+            "name": "entityId",
+            "type": "i32"
+          },
+          {
+            "name": "type",
+            "type": "i8"
+          },
+          {
+            "name": "position",
+            "type": "position_iii"
+          }
+        ]
+      ],
+      "packet_open_window": [
+        "container",
+        [
+          {
+            "name": "windowId",
+            "type": "i8"
+          },
+          {
+            "name": "inventoryType",
+            "type": "i8"
+          },
+          {
+            "name": "windowTitle",
+            "type": "string"
+          },
+          {
+            "name": "slotCount",
+            "type": "i8"
+          },
+          {
+            "name": "useProvidedTitle",
+            "type": "bool"
+          },
+          {
+            "name": "entityId",
```

Comment:
This is conditional on inventoryType equaling 11 according to the source other wise this field doesn't exist

Reply:
Should I remove the field or add a conditional?

Reply:
It should be conditional using a switch I believe.